README.md
# apoplexy.org
Official Website


ChangeLog.txt
v3.1 (February 7, 2018) - Norbert de Jonge
--------------------
Back from 'retirement' to implement two frequently requested features.
+ When editing PoP1 for DOS, a separate Map window can now be opened ("m"), that displays a movable and zoomable abstract representation of the entire level, which highlights related tiles when hovering over buttons, gates, exits left, loose tiles, etc.
+ PoP1 for DOS guard details can now be modified via the EXE screen (F2). This makes it easy to, for example, use guards with different amounts of hit points in the same level.
* Minor bug fixes.

v3.0 (February 4, 2016) - Norbert de Jonge
--------------------
This is the - or at least my - final version of apoplexy. It's time to move on. I enjoyed working on the program and I've learned a lot through the years. This program is free software (libre, also gratis), so feel free to redistribute and/or modify it without my permission.

v3.0 RC3 (January 15, 2016) - Norbert de Jonge
--------------------
* Updated SDLPoP support to v1.16.
* SNES playtesting now also skips the main menu.
* Slightly changed the Compress() function.
* The Windows port is now also 64-bit.
- Removed the "AMD64 32-bit" SDLPoP text.

v3.0 RC2 (December 21, 2015) - Norbert de Jonge
--------------------
+ Added support for SDLPoP (v1.16b4).
* Fixed two minor bugs (teleport marker, sprite zooming).

v3.0 RC1 (December 13, 2015) - Norbert de Jonge
--------------------
+ The program can now also edit PoP1 for SNES levels!
* Minor bug fixes.
This is a code complete release candidate. Version 3.0 will be released on February 4th, 2016. If you run into any bugs or defects with this release candidate, please report them before February.

v2.7 (August 21, 2015) - Norbert de Jonge
--------------------
+ Now auto-enables all PoP1 resources, if necessary (and possible).
+ For PoP1, the "0"-"9" keys are now shortcuts for various tiles.
+ The "i" key now toggles tile information.
* Improved several of the PoP1 spike images.
* Cleaned up the code a bit (e.g. simplified PreLoadSet()).
* Less #define's are used for the calculation of tile offsets.
* Changed various iEditPoP if/else statements to switches.

v2.6b (September 17, 2014) - Norbert de Jonge
--------------------
+ Expanded the screen where PoP1's PRINCE.EXE can be changed.[1]
* Better handling of PoP2 (static and dynamic) guards with unexpected values.
[1] Users can now also change whether the prince has his sword (default, never, always), where the prince wins (level, room), and the environments and enemy resources used for each level.

v2.5.1 (September 13, 2014) - Norbert de Jonge
--------------------
* Changed PoP2's "spawn if guard(s) alive" to "prince row".

v2.5 (August 11, 2014) - Norbert de Jonge
--------------------
+ Added (optional, togglable) fullscreen support.
+ Added a screen where PoP1's PRINCE.EXE can be changed.
* Fixed a minor event-related bug.

v2.4 (August 1, 2014) - Norbert de Jonge
--------------------
+ Levels can now also be imported from XML.
* Swapped the guard directions in the XML output.

v2.3.1 (July 30, 2014) - Norbert de Jonge
--------------------
* Fixed the XML output format (empty-element tags).

v2.3 (July 6, 2014) - Norbert de Jonge
--------------------
+ Added (Xbox) game controller support.
* Fixed two minor bugs (PoP1 guard selection typo, black rectangle scaling).
* Included executable now 64-bit.

v2.2b (June 1, 2014) - Norbert de Jonge
--------------------
* Workaround for SDL bug #2274.
* Freeing message surfaces to prevent memory usage increase.
* When zoomed, properly displays large ruins and temple backgrounds.
* Fixed a mouse with interface buttons interaction problem.

v2.1b (February 1, 2014) - Norbert de Jonge
--------------------
* Migrated from SDL 1.2 to 2.0.
+ Added several additional gcc warning options in the Makefile.
+ The cursor now changes on the loading screen and when hovering over web links.
* Improved the code quite a bit after receiving new gcc warnings (see above) and after using cppcheck and clang for the first time.

v2.0 (December 28, 2013) - Norbert de Jonge
--------------------
+ The program can now also edit Prince of Persia 2 levels!
+ Some tiles in the editor are now animated.
+ Better loading screen, including animations.
+ Now displays a warning when hovering over marked tiles.
* Moved the images to various subdirectories.

v1.9b (September 20, 2013) - Norbert de Jonge
--------------------
+ Added a screen that can be used to place custom tiles.
+ It's now possible to double the interface size.
+ Users can now copy and paste rooms, and flip rooms (horizontally and vertically).
+ Levels can now also be exported to XML.
+ The events screen now displays the target tile.
+ It's now possible to sprinkle levels with random decorations.
+ Added a simple help screen.
+ Added two extra guard levels.
+ A LEVELS.BAK file is now being created before saving a level.
+ The application can now immediately display any level on startup.
+ It's now possible to switch between dungeon and palace environments.
* Several aesthetic and interaction improvements.[1]
* Clearing a room now also removes the guard.
* It's now easier to re-use the last used tile.
* The scroll wheel can now also be used for horizontal scrolling.
* Immediately playtesting specific levels now also works with non-default cheat codes.
* Randomize no longer uses - now marked - distorting tiles.
* It's now easier to decrease room numbers on the broken room links screen.
* Fixed the image of one of the palace tiles.
[1] Displays the room under the current room. If there is no room to the left, displays a wall. Better response to the selection of persons. Key repeat on. Less unnecessary screen updates.

v1.2b (July 16, 2012) - Norbert de Jonge
--------------------
+ Added a screen that can be used to (deliberately) create, change and fix so-called broken room links.
* Fixed a minor bug that sometimes caused unintentional changes to events.


t/index.php
<?php
/* apoplexy.org
 * Copyright (C) The apoplexy Team <info@apoplexy.org>
 *
 * This program is free software: you can redistribute it and/or modify it
 * under the terms of the GNU General Public License as published by the Free
 * Software Foundation, either version 3 of the License, or (at your option)
 * any later version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
 * more details.
 *
 * You should have received a copy of the GNU General Public License along with
 * this program. If not, see [ www.gnu.org/licenses/ ].
 */

include ('../apoplexy_def.php');
HTMLStart ('Tutorials');
Menu ('Tutorials');
print ('<h1>Tutorials</h1>');
print ('<div class="div_content">');

/*****************************************************************************/
function VideoWebM ($sFile, $sTitle, $sDate)
/*****************************************************************************/
{
print ('
<h2>' . $sTitle . ' <span style="font-size:20px;">(' . $sDate . ')</span></h2>
<video src="/tutorials/' . $sFile . '" controls preload="metadata" style="width:100%;">Your browser does not properly support the HTML5 video tag.</video>
');
}
/*****************************************************************************/
function VideoYT ($sCode, $sTitle)
/*****************************************************************************/
{
	print ('<a target="_blank" href="https://www.youtube.com/watch?v=' . $sCode . '">' . $sTitle . '</a>');
}
/*****************************************************************************/
?>

<div class="container-fluid">
<div class="row">
<div class="col-sm-6">

<span class="span_tutorialh">Official Videos</span>
<?php
	VideoWebM ('apoplexy_1_1.webm', 'Instructional Video', '30 January 2011');
	VideoWebM ('apoplexy_1_9b.webm', 'Supplement 1', '20 September 2013');
	VideoWebM ('apoplexy_3_0_RC1.webm', 'Supplement 2', '13 December 2015');
?>

</div>
<div class="col-sm-6" style="text-align:center;">

<span class="span_tutorialh">Various</span>
<h2>Documents</h2>
<span class="span_tutoriall">Special Events</span>
<p>
<a href="http://www.popot.org/documentation/documents/2015-08-07_PoP1_Special_Events.pdf">PoP1 for DOS</a>
<br>
<a href="http://www.popot.org/documentation/documents/2014-07-10_PoP2_Special_Events.pdf">PoP2 for DOS</a>
</p>
<h2>Community Videos</h2>
<span class="span_tutoriall">by Aram Sev.</span>
<p>
<?php VideoYT ('zA6kbNRleiI', 'quickly build a level'); ?>
<br>
<?php VideoYT ('6JF2gayyI_w', 'Prince test - quickly create a mod'); ?>
<br>
<?php VideoYT ('n1o3sAI0VYs', 'How to use the new 3.1 (part 1)'); ?>
<br>
<?php VideoYT ('fitfN32CkW4', 'How to use the new 3.1 (part 2)'); ?>
<br>
<?php VideoYT ('7sJ5rKTUk4U', 'only Prince of Persia 1'); ?>
</p>
<span class="span_tutoriall">by yaqxsw</span>
<p>
<?php VideoYT ('wumqeXAv_Aw', '1. "For beginners"'); ?>
<br>
<?php VideoYT ('XrbF5ujwymk', '2. "Create gates"'); ?>
<br>
<a target="_blank" href="https://www.youtube.com/watch?v=51Du9x1kDWk">3. "Blue potion"</a>
<br>
<a target="_blank" href="https://www.youtube.com/watch?v=ZxrOWyCIXbY">4. "Creating broken room"</a>
<br>
<a target="_blank" href="https://www.youtube.com/watch?v=Jouzt7EGszI">5. "Create Time, Texture, Life!!!"</a>
<br>
<a target="_blank" href="https://www.youtube.com/watch?v=N9HfuDeg8XY">6. "Modifier objects"</a>
<br>
<a target="_blank" href="https://www.youtube.com/watch?v=uW6NJkss8h0">7. "SDLPoP and how pass levelset"</a>
<br>
<a target="_blank" href="https://www.youtube.com/watch?v=P6aXc_SnmBI">8. "Change Graphic and Texture"</a>
<br>
<a target="_blank" href="https://www.youtube.com/watch?v=3Ho9C_sWHg8">9. "Screenshot and upload MOD"</a>
</p>

</div>
</div>
</div>

<?php
	print ('</div>');
	HTMLEnd();
?>


news_rss.php
<?php
/* apoplexy.org
 * Copyright (C) The apoplexy Team <info@apoplexy.org>
 *
 * This program is free software: you can redistribute it and/or modify it
 * under the terms of the GNU General Public License as published by the Free
 * Software Foundation, either version 3 of the License, or (at your option)
 * any later version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
 * more details.
 *
 * You should have received a copy of the GNU General Public License along with
 * this program. If not, see [ www.gnu.org/licenses/ ].
 */

	header ('content-type: text/xml');
	ini_set ('short_open_tag', 0);

	/*** Do not move this down, or short_open_tag will cause problems. ***/
	print ('<?xml version="1.0" encoding="UTF-8" ?>' . "\n");
?>
<!-- https://www.apoplexy.org/news_rss.php -->
<rss version="2.0" xml:base="https://www.apoplexy.org/" xmlns:atom="http://www.w3.org/2005/Atom">
	<channel>
		<title>Recent apoplexy News</title>
		<link>https://www.apoplexy.org/</link>
		<atom:link href="https://www.apoplexy.org/news_rss.php" rel="self" type="application/rss+xml" />
		<language>en-us</language>
		<description>A level editor of Prince of Persia 1 (for DOS and SNES) and 2 (for DOS).</description>
<?php
$xml = simplexml_load_file ('news.xml', 'SimpleXMLElement', LIBXML_NOCDATA);
$arXML = (array)$xml;
$iNews = 0;
foreach ($arXML['item'] as $key => $value)
{
	$sDate = strval ($value['date']);
	$sDateH = date ('F j, Y', strtotime ($sDate));
	$sText = strval ($value->html);
	$sRSSDate = date ('r', strtotime ($sDate));

	if ($iNews == 0)
	{
		print ("\t\t" . '<pubDate>' . $sRSSDate . '</pubDate>' . "\n");
		print ("\t\t" . '<lastBuildDate>' . $sRSSDate . '</lastBuildDate>' . "\n");
	}

	print ("\t\t" . '<item>' . "\n");
	print ("\t\t\t" . '<title>News for ' . $sDateH . '</title>' . "\n");
	print ("\t\t\t" . '<link>https://www.apoplexy.org/#' . $sDate . '</link>' . "\n");
	print ("\t\t\t" . '<description><![CDATA[' . $sText . ']]></description>' . "\n");
	print ("\t\t\t" . '<pubDate>' . $sRSSDate . '</pubDate>' . "\n");
	print ("\t\t\t" . '<guid>https://www.apoplexy.org/#' . $sDate . '</guid>' . "\n");
	print ("\t\t" . '</item>' . "\n");

	$iNews++;
	if ($iNews == 10) { break; }
}
?>
	</channel>
</rss>


apoplexy_def.php
<?php
/* apoplexy.org
 * Copyright (C) The apoplexy Team <info@apoplexy.org>
 *
 * This program is free software: you can redistribute it and/or modify it
 * under the terms of the GNU General Public License as published by the Free
 * Software Foundation, either version 3 of the License, or (at your option)
 * any later version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
 * more details.
 *
 * You should have received a copy of the GNU General Public License along with
 * this program. If not, see [ www.gnu.org/licenses/ ].
 */

error_reporting (-1);
ini_set ('display_errors', 'On');

ini_set ('output_buffering', 'Off');
while (@ob_end_flush());

date_default_timezone_set ('UTC');

ini_set ('php.internal_encoding', 'UTF-8');
mb_internal_encoding ('UTF-8');

/*****************************************************************************/
function HTMLStart ($sTitle)
/*****************************************************************************/
{
print ('
<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1">
<!-- Always start with the above 3 meta tags. -->
');
if ($sTitle == 'News')
{
	print ('<meta name="description" content="A level editor of Prince of Persia 1 (for DOS and SNES) and 2 (for DOS).">');
}
print ('
<meta name="keywords" content="apoplexy, level editor, Prince of Persia, DOS, SNES, free software, open source, Linux, Windows">
<meta name="author" content="">
<link rel="icon" href="/images/favicon_16x16.png" sizes="16x16" type="image/png">
<link rel="icon" href="/images/favicon_32x32.png" sizes="32x32" type="image/png">
<link rel="alternate" type="application/rss+xml" title="Recent apoplexy News" href="/news_rss.php">
<title>' . $sTitle . ' · apoplexy</title>

<!-- CSS -->
<link rel="stylesheet" type="text/css" href="/bootstrap/css/bootstrap.min.css">
<link rel="stylesheet" type="text/css" href="/css/gh-fork-ribbon.css">
<link rel="stylesheet" type="text/css" href="/apoplexy.css?v=1">

<!-- JS -->
<script src="jquery.js"></script>
<script src="bootstrap/js/bootstrap.min.js"></script>
<script src="js/ie10-viewport-bug-workaround.js"></script>
<!--[if lt IE 9]>
<script src="js/html5shiv.min.js"></script>
<script src="js/respond.min.js"></script>
<![endif]-->
<script src="apoplexy.js?v=1"></script>
</head>
<body>
<a target="_blank" href="https://github.com/apoplexy?tab=repositories" class="github-fork-ribbon" data-ribbon="Fork me on GitHub" title="Fork me on GitHub">Fork me on GitHub</a>
<a href="/"><div style="background-color:#6e88a8; text-align:center; border-bottom:2px solid #000;">
<img src="/images/header_01.png" alt="apoplexy text" style="max-height:100px; max-width:100%;">
</div></a>
<div class="container-fluid">
<div class="row">
<div class="visible-lg col-lg-2"></div>
<div class="col-lg-8 col-md-12 div_main">
');
}
/*****************************************************************************/
function HTMLEnd ()
/*****************************************************************************/
{
print ('
</div>
<div class="visible-lg col-lg-2"></div>
</div>
</div>

<footer class="footer">
<div class="container-fluid">
<div class="row">
<div class="visible-lg col-lg-2"></div>
<div class="col-lg-8 col-md-12" style="padding:0 20px;">
<p class="footertext">&copy; apoplexy Team | <a target="_blank" href="https://validator.w3.org/check?uri=referer"><img src="/images/W3C_HTML5.png" alt="W3C HTML5"></a> | <a target="_blank" href="https://en.wikipedia.org/wiki/Responsive_web_design"><img src="/images/RWD.png" alt="Responsive web design" title="Responsive web design"></a></p>
</div>
<div class="visible-lg col-lg-2"></div>
</div>
</div>
</footer>

</body>
</html>
');
}
/*****************************************************************************/
function Menu ($sActive)
/*****************************************************************************/
{
	$arMenu = array();
	$arMenu['Download'] = '/d/';
	$arMenu['Screenshots'] = '/s/';
	$arMenu['Tutorials'] = '/t/';
	$arMenu['Contact &amp; Links'] = '/c/';

	print ('<div class="container-fluid">');
	print ('<div class="row div_menu">');
	foreach ($arMenu as $key => $value)
	{
		print ('<div class="col-lg-3 col-xs-6"><a href="' . $value .
			'" class="a_menu');
		if ($key == $sActive) { print (' active'); }
		print ('">' . $key . '</a></div>' . "\n");
	}
	print ('</div>');
	print ('</div>');
}
/*****************************************************************************/
?>


d/index.php
<?php
/* apoplexy.org
 * Copyright (C) The apoplexy Team <info@apoplexy.org>
 *
 * This program is free software: you can redistribute it and/or modify it
 * under the terms of the GNU General Public License as published by the Free
 * Software Foundation, either version 3 of the License, or (at your option)
 * any later version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
 * more details.
 *
 * You should have received a copy of the GNU General Public License along with
 * this program. If not, see [ www.gnu.org/licenses/ ].
 */

include ('../apoplexy_def.php');

$GLOBALS['last_release'] = 20;

/*** $iRelease ***/
$iRelease = 0;
if (isset ($_GET['r']))
{
	$iRelease = intval ($_GET['r']);
	if (($iRelease < 1) || ($iRelease > $GLOBALS['last_release']))
		{ $iRelease = 0; }
}

if ($iRelease == 0)
{
	HTMLStart ('Download');
	Menu ('Download');
	print ('<h1>Download apoplexy 3.2</h1>');
} else {
	SetRelease ($iRelease);
	HTMLStart ($GLOBALS['rtitleh1']);
	Menu ($GLOBALS['rtitleh1']);
	print ('<h1>' . $GLOBALS['rtitleh1'] . '</h1>');
}
print ('<div class="div_content">');

/*****************************************************************************/
function SetRelease ($iRelease)
/*****************************************************************************/
{
	switch ($iRelease)
	{
		case 20:
			$sVersion = '3.2';
			$GLOBALS['rdate'] = '2018-02-21';
$GLOBALS['rchanges'] = '+ If one or more games are not present, instead of merely showing clickable web links to allow the user to manually add the games, buttons can now be clicked to auto-download the games.
* Fixed a dest.x/y bug in ShowImageBasic().
* Moved the Windows port back to 32-bit.
* Various small changes to allow this to become a multi-developer project.[3]
* Added the project to GitHub, at https://github.com/apoplexy/apoplexy.
[3] Moved the code to src/, new domain (apoplexy.org) and website, new icon/logo, additional project files (compiling.txt, coding-style.txt, credits.txt).';
			break;
		case 19:
			$sVersion = '3.1';
			$GLOBALS['rdate'] = '2018-02-07';
$GLOBALS['rchanges'] = 'Back from \'retirement\' to implement two frequently requested features.
+ When editing PoP1 for DOS, a separate Map window can now be opened ("m"), that displays a movable and zoomable abstract representation of the entire level, which highlights related tiles when hovering over buttons, gates, exits left, loose tiles, etc.
+ PoP1 for DOS guard details can now be modified via the EXE screen (F2). This makes it easy to, for example, use guards with different amounts of hit points in the same level.
* Minor bug fixes.';
			break;
		case 18:
			$sVersion = '3.0';
			$GLOBALS['rdate'] = '2016-02-04';
$GLOBALS['rchanges'] = 'This is the - or at least my - final version of apoplexy. It\'s time to move on. I enjoyed working on the program and I\'ve learned a lot through the years. This program is free software (libre, also gratis), so feel free to redistribute and/or modify it without my permission.';
			break;
		case 17:
			$sVersion = '3.0 RC3';
			$GLOBALS['rdate'] = '2016-01-15';
$GLOBALS['rchanges'] = '* Updated SDLPoP support to v1.16.
* SNES playtesting now also skips the main menu.
* Slightly changed the Compress() function.
* The Windows port is now also 64-bit.
- Removed the "AMD64 32-bit" SDLPoP text.';
			break;
		case 16:
			$sVersion = '3.0 RC2';
			$GLOBALS['rdate'] = '2015-12-21';
$GLOBALS['rchanges'] = '+ Added support for SDLPoP (v1.16b4).
* Fixed two minor bugs (teleport marker, sprite zooming).';
			break;
		case 15:
			$sVersion = '3.0 RC1';
			$GLOBALS['rdate'] = '2015-12-13';
$GLOBALS['rchanges'] = '+ The program can now also edit PoP1 for SNES levels!
* Minor bug fixes.
This is a code complete release candidate. Version 3.0 will be released on February 4th, 2016. If you run into any bugs or defects with this release candidate, please report them before February.';
			break;
		case 14:
			$sVersion = '2.7';
			$GLOBALS['rdate'] = '2015-08-21';
$GLOBALS['rchanges'] = '+ Now auto-enables all PoP1 resources, if necessary (and possible).
+ For PoP1, the "0"-"9" keys are now shortcuts for various tiles.
+ The "i" key now toggles tile information.
* Improved several of the PoP1 spike images.
* Cleaned up the code a bit (e.g. simplified PreLoadSet()).
* Less #define\'s are used for the calculation of tile offsets.
* Changed various iEditPoP if/else statements to switches.';
			break;
		case 13:
			$sVersion = '2.6b';
			$GLOBALS['rdate'] = '2014-09-17';
$GLOBALS['rchanges'] = '+ Expanded the screen where PoP1\'s PRINCE.EXE can be changed.[1]
* Better handling of PoP2 (static and dynamic) guards with unexpected values.
[1] Users can now also change whether the prince has his sword (default, never, always), where the prince wins (level, room), and the environments and enemy resources used for each level.';
			break;
		case 12:
			$sVersion = '2.5.1';
			$GLOBALS['rdate'] = '2014-09-13';
$GLOBALS['rchanges'] = '* Changed PoP2\'s "spawn if guard(s) alive" to "prince row".';
			break;
		case 11:
			$sVersion = '2.5';
			$GLOBALS['rdate'] = '2014-08-11';
$GLOBALS['rchanges'] = '+ Added (optional, togglable) fullscreen support.
+ Added a screen where PoP1\'s PRINCE.EXE can be changed.
* Fixed a minor event-related bug.';
			break;
		case 10:
			$sVersion = '2.4';
			$GLOBALS['rdate'] = '2014-08-01';
$GLOBALS['rchanges'] = '+ Levels can now also be imported from XML.
* Swapped the guard directions in the XML output.';
			break;
		case 9:
			$sVersion = '2.3.1';
			$GLOBALS['rdate'] = '2014-07-30';
$GLOBALS['rchanges'] = '* Fixed the XML output format (empty-element tags).';
			break;
		case 8:
			$sVersion = '2.3';
			$GLOBALS['rdate'] = '2014-07-06';
$GLOBALS['rchanges'] = '+ Added (Xbox) game controller support.
* Fixed two minor bugs (PoP1 guard selection typo, black rectangle scaling).
* Included executable now 64-bit.';
			break;
		case 7:
			$sVersion = '2.2b';
			$GLOBALS['rdate'] = '2014-06-01';
$GLOBALS['rchanges'] = '* Workaround for SDL bug #2274.
* Freeing message surfaces to prevent memory usage increase.
* When zoomed, properly displays large ruins and temple backgrounds.
* Fixed a mouse with interface buttons interaction problem.';
			break;
		case 6:
			$sVersion = '2.1b';
			$GLOBALS['rdate'] = '2014-02-01';
$GLOBALS['rchanges'] = '* Migrated from SDL 1.2 to 2.0.
+ Added several additional gcc warning options in the Makefile.
+ The cursor now changes on the loading screen and when hovering over web links.
* Improved the code quite a bit after receiving new gcc warnings (see above) and after using cppcheck and clang for the first time.';
			break;
		case 5:
			$sVersion = '2.0';
			$GLOBALS['rdate'] = '2013-12-28';
$GLOBALS['rchanges'] = '+ The program can now also edit Prince of Persia 2 levels!
+ Some tiles in the editor are now animated.
+ Better loading screen, including animations.
+ Now displays a warning when hovering over marked tiles.
* Moved the images to various subdirectories.';
			break;
		case 4:
			$sVersion = '1.9b';
			$GLOBALS['rdate'] = '2013-09-20';
$GLOBALS['rchanges'] = '+ Added a screen that can be used to place custom tiles.
+ It\'s now possible to double the interface size.
+ Users can now copy and paste rooms, and flip rooms (horizontally and vertically).
+ Levels can now also be exported to XML.
+ The events screen now displays the target tile.
+ It\'s now possible to sprinkle levels with random decorations.
+ Added a simple help screen.
+ Added two extra guard levels.
+ A LEVELS.BAK file is now being created before saving a level.
+ The application can now immediately display any level on startup.
+ It\'s now possible to switch between dungeon and palace environments.
* Several aesthetic and interaction improvements.[1]
* Clearing a room now also removes the guard.
* It\'s now easier to re-use the last used tile.
* The scroll wheel can now also be used for horizontal scrolling.
* Immediately playtesting specific levels now also works with non-default cheat codes.
* Randomize no longer uses - now marked - distorting tiles.
* It\'s now easier to decrease room numbers on the broken room links screen.
* Fixed the image of one of the palace tiles.
[1] Displays the room under the current room. If there is no room to the left, displays a wall. Better response to the selection of persons. Key repeat on. Less unnecessary screen updates.';
			break;
		case 3:
			$sVersion = '1.2b';
			$GLOBALS['rdate'] = '2012-07-16';
$GLOBALS['rchanges'] = '+ Added a screen that can be used to (deliberately) create, change and fix so-called broken room links.
* Fixed a minor bug that sometimes caused unintentional changes to events.';
			break;
		case 2:
			$sVersion = '1.1';
			$GLOBALS['rdate'] = '2011-01-30';
$GLOBALS['rchanges'] = '+ All actions can now also be performed with the keyboard. The keyboard shortcuts are listed in docs/keys.txt, and they\'re also shown when running the application. In order to ensure consistency, the "h/j/u/n keys" are not used on the main screen.
+ On the room links screen, a room that is being moved is now marked with red stripes.
+ Added seven additional interface sound effects.
+ The application now warns the user about unsaved changes.
* Changed popup.wav (3.7MB to 6kB) in order to decrease the total size of the package.
* There\'s now always a tile selected, to allow for faster switching between mouse and keyboard.
* Cleaned up the code a bit, mostly by adding more functions.
* The Princed Resources editor has been moved into the pr/ directory.';
			break;
		case 1:
			$sVersion = '1.0b';
			$GLOBALS['rdate'] = '2008-04-03';
$GLOBALS['rchanges'] = '+ Initial release.';
			break;
	}
	$GLOBALS['rtitleh1'] = 'ChangeLog Version ' . $sVersion;
}
/*****************************************************************************/
function DownAndNews ()
/*****************************************************************************/
{
print ('
<span style="display:block; float:left; width:calc(50% - 10px); text-align:center; margin-right:10px;">
<span style="display:block; margin:16px 0;">for Windows</span>
<img src="/images/Windows.png" alt="Windows">
<a href="/releases/apoplexy-3.2-win32.zip" class="download">Download Now</a>
</span>
<span style="display:block; float:left; width:calc(50% - 10px); text-align:center; margin-left:10px;">
<span style="display:block; margin:16px 0;">for GNU/Linux</span>
<img src="/images/GNULinux.png" alt="GNU/Linux">
<a href="/releases/apoplexy-3.2.tar.gz" class="download">Download Now</a>
</span>
<span style="display:block; clear:both;"></span>
<span style="display:block; text-align:center;">A <a target="_blank" href="/ChangeLog.txt">changelog</a> is available.</span>

<h2>News</h2>
');

	$xml = simplexml_load_file ('../news.xml',
		'SimpleXMLElement', LIBXML_NOCDATA);
	$arXML = (array)$xml;
	foreach ($arXML['item'] as $key => $value)
	{
		$sDate = strval ($value['date']);
		$sDateH = date ('F j, Y', strtotime ($sDate));
		$sText = strval ($value->html);
		print ('<span id="' . $sDate . '" style="font-size:20px;">' . $sDateH . '</span>' . "\n");
		print ('<p>' . $sText . '</p>' . "\n");
	}
}
/*****************************************************************************/
function Release ($iRelease)
/*****************************************************************************/
{
	$sDateH = date ('F j, Y', strtotime ($GLOBALS['rdate']));
	print ('<p style="font-size:20px;"><a href="/d/">&laquo; Back</a></p>');
	print ('<h2>' . $GLOBALS['rtitleh1'] . '</h2>');
	print ('<img src="/images/apoplexy_32x32.png" style="float:left; padding:10px;">');
	print ('<span style="display:block; font-size:16px; margin-bottom:20px;">');
	print ('Released by <a target="_blank" href="https://www.popot.org/profile.php?user_id=1">Norbert</a> on ' . $sDateH . '.');
	print ('<br>');
	print ('<span class="italic">Thoughts? E-mail <span class="rtl">gro.yxelpopa@ofni</span> or leave a comment on our <a target="_blank" href="http://forum.princed.org/viewforum.php?f=112">forum</a>.</span>');
	print ('</span>');
	print ('<span class="changes">' . nl2br ($GLOBALS['rchanges']) . '</span>');
}
/*****************************************************************************/
?>

<div class="container-fluid">
<div class="row">
<div class="col-sm-6">
<p style="text-align:justify;">A level editor of <span style="font-style:italic;">Prince of Persia</span> 1 (for DOS and SNES) and 2 (for DOS). This program is free software: you can redistribute it and/or modify it under the terms of the <a target="_blank" href="https://www.gnu.org/licenses/gpl-3.0.en.html">GNU General Public License</a> as published by the <a target="_blank" href="https://www.fsf.org/">Free Software Foundation</a>, either version 3 of the License, or (at your option) any later version.</p>
<?php
	if ($iRelease != 0)
	{
		Release ($iRelease);
	} else {
		DownAndNews();
	}
?>
</div>
<div class="col-sm-6">

<div class="polaroid rotate_right">
	<a target="_blank" href="/screenshots/apoplexy-3.1_PoP1_DOS.png">
		<img src="/screenshots/apoplexy-3.1_PoP1_DOS.png" alt="PoP1 for DOS">
	</a>
	<p class="caption">Editing <i>Prince of Persia 1</i> for DOS</p>
</div>
<span style="display:block; clear:both;"></span>
<div class="polaroid rotate_left">
	<a target="_blank" href="/screenshots/apoplexy-3.1_PoP2_DOS.png">
		<img src="/screenshots/apoplexy-3.1_PoP2_DOS.png" alt="PoP2 for DOS">
	</a>
	<p class="caption" style="text-align:right;">Editing <i>Prince of Persia 2</i> for DOS</p>
</div>
<span style="display:block; clear:both;"></span>
<div class="polaroid rotate_right">
	<a target="_blank" href="/screenshots/apoplexy-3.1_PoP1_SNES.png">
		<img src="/screenshots/apoplexy-3.1_PoP1_SNES.png" alt="PoP1 for SNES">
	</a>
	<p class="caption">Editing <i>Prince of Persia 1</i> for SNES</p>
</div>

</div>
</div>
</div>

<?php
	print ('</div>');
	HTMLEnd();
?>
